### ISSUE TITLE
Activation of tracking only after user consent

### ISSUE THREAD

**Issue by kaboume** [johannschopplich/nuxt-gtag#9]
### Describe the feature

Good morning,
Thanks for this plugin.
I use it on a website in France, a country in which we need to ask the user's consent, via the display of a banner, to trace his navigation on the site.
I was wondering how by default not to activate the tracking and to be able to activate it only if the user clicks on the button of the banner.
Does the plugin allow this?
THANKS.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Can you think of other implementations of this feature?

### Final checks

- [X] Check existing [issues](https://github.com/johannschopplich/nuxt-gtag/issues).

**Comment by anirudhge** [comment]
Anyone have a solution to this?

**Comment by johannschopplich** [comment]
@MrGeminus, @PiotrGrobelak, @Ethan-Hill, @russback, @anirudhge I have added support for manual consent management:

If you want to disable tracking by default, you can set the `initialConsent` option to `false`. This will prevent the `gtag.js` script from loading until the user has consented to tracking.

```ts
export default defineNuxtConfig({
  modules: ['nuxt-gtag'],

  gtag: {
    id: 'G-XXXXXXXXXX',
    initialConsent: false
  }
})
```

To manually manage consent, you can use the [`useGtagConsent` composable](https://github.com/johannschopplich/nuxt-gtag#usegtagconsent) to set the consent state, e.g. after the user has accepted your cookie policy.

```vue
<script setup lang="ts">
function acceptTracking() {
  useGtagConsent(true)
}
</script>

<template>
  <button @click="acceptTracking">
    Accept Tracking
  </button>
</template>
```

Available in `nuxt-gtag` v0.5.0+.

**Comment by anirudhge** [comment]
This is great, thank you so much! 

I do want to note though, that it is possible for GA4 to do cookie-less tracking. When cookies are rejected, gtag will still send pings to GA4.

However, it doesn't seem possible to first set the consent options of the user, before loading the tag? This means that we first have to load gtag, which sets the cookies, before sending the consent.

**Comment by anirudhge** [comment]
Unfortunately for me the  ` useGtagConsent(true)` composable doesn't seem to do anything. I made sure to run it client-side.

**Comment by johannschopplich** [comment]
@anirudhge You're right, there is indeed a bug. Thanks for sharing!

**Comment by johannschopplich** [comment]
You're right, working on a fix... Please test again and if an error persists, open another ticket. Thanks!

**Comment by russback** [comment]
@johannschopplich thanks for the update. My config looks like this (I am using  my runtime config and `.env` for the container ID):

```
gtag: {
  initialConsent: false
}
```

When I call `useGtagConsent(true)` when consent is given on button click (or in the layout template if consent has been cookied), I get the following reported from the Google Analytics Debugger Chrome dev tool:

```
Processing GTAG command: ["config", "G-Z6D9QQFKZ4", ""]
Ignored "config" command. Invalid arguments found.
Expected: "[config, string, plain Object]"Actual:   ["config", "G-Z6D9QQFKZ4", ""]
```

What format should I be defining here whilst still using `.env` for the container ID?
